### PR TITLE
Integrating entity:applyOffsetAngForce and array:massCenter

### PR DESCRIPTION
**Pull request by Starkkz**
So I'm bringing those two E2 functions for people to be able to move contraptions from within their real gravity centers. I didn't know where the E2 description for this file was so I didn't add it.

entity:applyOffsetAngForce(angle angForce, vector position)
vector = array:massCenter()

### CODE REVIEW

**Comment by Nebual** [comment]
Nice, nice. It's a bit of a hassle to calculate that within E2, so I approve of the inclusion of these convenience functions. I think that finishes the thrust math abstraction layer.

**Comment by Starkkz** [comment]
Oh well, I wanted to add inertia and angular velocity but I didn't manage to calculate them, I couldn't find any topics about that either.

**Comment by Divran** [comment]
You need to use `check(position)` too

File: lua/entities/gmod_wire_expression2/core/entity.lua [hunk 1]
```diff
@@ -548,6 +585,24 @@ e2function vector entity:inertia()
 	return this:GetPhysicsObject():GetInertia()
 end
 
+e2function vector array:massCenter()
+	if !next(this) then return {0, 0, 0} end
```
Comment (Divran) [inline lua/entities/gmod_wire_expression2/core/entity.lua]: would prefer to use standard lua syntax, ie `not`

**Comment by Starkkz** [comment]
I just found this.
http://www.wiremod.com/forum/expression-2-discussion-help/28660-help-angle-manipulation-2.html
Can I get some more time to recreate the array:inertia() function in Lua?

**Comment by TomyLobo** [comment]
@Divran can you extract a helper function so we dont have the entire angforce code twice?
If you can, make it tail-recurse for efficiency and such.

Also, while you're at it, add applyOffsetTorque to match maybe?

File: lua/entities/gmod_wire_expression2/core/entity.lua [hunk 2]
```diff
@@ -511,6 +511,44 @@ e2function void entity:applyAngForce(angle angForce)
 	end
 end
 
+e2function void entity:applyOffsetAngForce(angle angForce, vector position)
```
Comment (TomyLobo) [inline lua/entities/gmod_wire_expression2/core/entity.lua]: Unless that's a local position and not a world-grid offset, that 2nd parameter should be named "offset", not "position".

**Pull request closed without merging**